lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -134,20 +132,25 @@ void amf_ue_context_update_coll_keys(
   ue_context_p->amf_teid_n11 = amf_teid_n11;
 
   if (guti_p) {
-    if ((guti_p->guamfi.amf_set_id !=
-         ue_context_p->amf_context.m5_guti.guamfi.amf_set_id) ||
-        (guti_p->guamfi.amf_regionid !=
-         ue_context_p->amf_context.m5_guti.guamfi.amf_regionid) ||
-        (guti_p->m_tmsi != ue_context_p->amf_context.m5_guti.m_tmsi) ||
-        (guti_p->guamfi.plmn.mcc_digit1 !=
-         ue_context_p->amf_context.m5_guti.guamfi.plmn.mcc_digit1) ||
-        (guti_p->guamfi.plmn.mcc_digit2 !=
-         ue_context_p->amf_context.m5_guti.guamfi.plmn.mcc_digit2) ||
-        (guti_p->guamfi.plmn.mcc_digit3 !=
-         ue_context_p->amf_context.m5_guti.guamfi.plmn.mcc_digit3) ||
-        (ue_context_p->amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID)) {
-      m_rc = amf_ue_context_p->guti_ue_context_htbl.remove(*guti_p);
+    if (((guti_p->guamfi.amf_set_id !=
+          ue_context_p->amf_context.m5_guti.guamfi.amf_set_id) ||
+         (guti_p->guamfi.amf_regionid !=
+          ue_context_p->amf_context.m5_guti.guamfi.amf_regionid) ||
+         (guti_p->m_tmsi != ue_context_p->amf_context.m5_guti.m_tmsi) ||
+         (guti_p->guamfi.plmn.mcc_digit1 !=
+          ue_context_p->amf_context.m5_guti.guamfi.plmn.mcc_digit1) ||
+         (guti_p->guamfi.plmn.mcc_digit2 !=
+          ue_context_p->amf_context.m5_guti.guamfi.plmn.mcc_digit2) ||
+         (guti_p->guamfi.plmn.mcc_digit3 !=
+          ue_context_p->amf_context.m5_guti.guamfi.plmn.mcc_digit3)) ||
+        (ue_context_p->amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID) &&
+            ((guti_p->guamfi.amf_set_id != 0) &&
+             (guti_p->guamfi.amf_regionid != 0) && (guti_p->m_tmsi != 0) &&
+             (guti_p->guamfi.plmn.mcc_digit1 != 0) &&
+             (guti_p->guamfi.plmn.mcc_digit2 != 0) &&
+             (guti_p->guamfi.plmn.mcc_digit3 != 0))) {
       if (INVALID_AMF_UE_NGAP_ID != amf_ue_ngap_id) {
```

Comment:
Not related with this PR, but please fix the style here too
```suggestion
      if (amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID ) {
```

Reply:
Addressed Review Comment